### PR TITLE
Add check for unsupported regions in retry logic for aws_wellarchitected_workload table.

### PR DESCRIPTION
**Pull request by bigdatasourav**
# Example query results
<details>
  <summary>Results</summary>
  
```
> select * from aws_wellarchitected_workload
+---------------+------------------------------------------------------------------------------------------+----------------------------------+----------------------+---------------------+---------------+
| workload_name | workload_arn                                                                             | workload_id                      | architectural_design | description         | environment   |
+---------------+------------------------------------------------------------------------------------------+----------------------------------+----------------------+---------------------+---------------+
| fdf           | arn:aws:wellarchitected:us-east-1:533793682495:workload/873e847292431181e0c2b2489e36a903 | 873e847292431181e0c2b2489e36a903 | <null>               | ffewfrfwfdfregfregr | PREPRODUCTION |
+---------------+------------------------------------------------------------------------------------------+----------------------------------+----------------------+---------------------+---------------+

> select * from aws_wellarchitected_workload where workload_id = '873e847292431181e0c2b2489e36a903'
+---------------+------------------------------------------------------------------------------------------+----------------------------------+----------------------+---------------------+---------------+
| workload_name | workload_arn                                                                             | workload_id                      | architectural_design | description         | environment   |
+---------------+------------------------------------------------------------------------------------------+----------------------------------+----------------------+---------------------+---------------+
| fdf           | arn:aws:wellarchitected:us-east-1:533793682495:workload/873e847292431181e0c2b2489e36a903 | 873e847292431181e0c2b2489e36a903 | <null>               | ffewfrfwfdfregfregr | PREPRODUCTION |
+---------------+------------------------------------------------------------------------------------------+----------------------------------+----------------------+---------------------+---------------+

```
</details>

### CODE REVIEW

**Review by rajlearner17 — COMMENTED** [review]
@bigdatasourav As discussed we need to have some design discussion on how to handle this error/exception. Should we throw an error, that may risk suppressing the regions with data.

File: aws/service.go [hunk 1]
```diff
@@ -1878,6 +1878,13 @@ func (r ConnectionErrRetryer) ShouldRetry(req *request.Request) bool {
 				if strings.Contains(awsErr.OrigErr().Error(), "http://169.254.169.254/latest") && req.RetryCount > 3 {
 					return false
 				}
+
+				// AWS Well-Architected Tool is not supported in all region yet. For unsupported regions It throws err
```
Comment (cbruno10) [inline aws/service.go:1882]:
```suggestion
				// AWS Well-Architected Tool is not supported in all regions. For unsupported regions the API throws an error, e.g.,
```

File: aws/service.go [hunk 2]
```diff
@@ -1878,6 +1878,13 @@ func (r ConnectionErrRetryer) ShouldRetry(req *request.Request) bool {
...
+				// Post "https://wellarchitected.ap-northeast-3.amazonaws.com/workloadsSummaries": dial tcp: lookup wellarchitected.ap-northeast-3.amazonaws.com: no such host
+				// https://aws.amazon.com/about-aws/whats-new/2020/04/aws-well-architected-tool-available-additional-regions/
```
Comment (cbruno10) [inline aws/service.go:1884]: Is there a better link we can show? This is from 2020, and is bound to change over time. Is there a page that has a list of currently supported regions?

**Review by rajlearner17 — APPROVED** [review]
Looks good.

**Review by cbruno10 — COMMENTED** [review]
Please see comments in https://github.com/turbot/steampipe-plugin-aws/pull/862#pullrequestreview-860961215, as I think they may apply here as well, thanks!

**Pull request merged**